### ISSUE TITLE
Use a `.cake` folder by default instead of a `tools` folder

### ISSUE THREAD

**Issue by augustoproiete** [cake-build/cake#3263]
> Tracking improvement originally from discussion https://github.com/cake-build/cake/discussions/2979 targeting Cake v2.0

**Comment by joshuacookdev** [comment]
Implementation question:

`./.cake/tools/` or `./.cake/`? I know the request says `./.cake/`, but I feel like putting the tools under `./.cake/tools/` might be a better long-term option. 

If no one has picked this up by the weekend, I'll likely play around with it - just wanted to bring that up for discussion before any work began on the issue.

**Comment by augustoproiete** [comment]
Hey @joshuacookdev,

TL;DR; I think `./.cake/tools/` is a good idea and probably where we'd want to get to, but let's do that as a separate issue to make it easier to review, test, & merge this one.

---

Long version:

At the moment, if you have (for example) a Cake script like the following:

```csharp
#addin "nuget:?package=Cake.ExcelDnaPack&version=1.0.1"
#module "nuget:?package=Cake.BuildSystems.Module&version=3.0.3"
#tool "nuget:?package=7-Zip.CommandLine"
#load "nuget:?package=Cake.Recipe&version=2.2.1"
```

You'll end up with with a folder structure like this:

![image](https://user-images.githubusercontent.com/177608/116961015-860d3e80-ac78-11eb-929d-7ed2d4ddf439.png)

As you can see, Cake `#addin`s and `#module`s get their own folder, but `#tool`s and `#recipe`s don't, and their packages are extracted to the root of the configured `tools` folder.

I think it would make sense to have `#tool`s and `#recipe`s in their own folders, but that might have a bigger impact and require a more thorough testing, and probably make this PR harder to review and merge.

I suggest we keep the scope of this one small, just renaming the default `tools` folder to `.cake` and we create two separate issues to be implemented after this one is merged:
- Change the default folder for Cake `#tool`s to `.cake/tools` instead of `.cake`
- Change the default folder for Cake `#recipe`s to `.cake/recipes` instead of `.cake`
